votor-messages/src/consensus_message.rs

```diff
@@ -0,0 +1,178 @@
+    pub fn new(certificate_type: CertificateType, slot: Slot, hash: Option<Hash>) -> Self {
+        match (certificate_type, hash) {
+            (CertificateType::Finalize, None) => Certificate::Finalize(slot),
+            (CertificateType::FinalizeFast, Some(hash)) => Certificate::FinalizeFast(slot, hash),
+            (CertificateType::Notarize, Some(hash)) => Certificate::Notarize(slot, hash),
+            (CertificateType::NotarizeFallback, Some(hash)) => {
+                Certificate::NotarizeFallback(slot, hash)
+            }
+            (CertificateType::Skip, None) => Certificate::Skip(slot),
+            _ => panic!("Invalid certificate type and hash combination"),
+        }
+    }
+
+    /// Get the certificate type
+    pub fn certificate_type(&self) -> CertificateType {
+        match self {
+            Certificate::Finalize(_) => CertificateType::Finalize,
+            Certificate::FinalizeFast(_, _) => CertificateType::FinalizeFast,
+            Certificate::Notarize(_, _) => CertificateType::Notarize,
+            Certificate::NotarizeFallback(_, _) => CertificateType::NotarizeFallback,
+            Certificate::Skip(_) => CertificateType::Skip,
+        }
+    }
+
+    /// Get the slot of the certificate
+    pub fn slot(&self) -> Slot {
+        match self {
+            Certificate::Finalize(slot)
+            | Certificate::FinalizeFast(slot, _)
+            | Certificate::Notarize(slot, _)
+            | Certificate::NotarizeFallback(slot, _)
+            | Certificate::Skip(slot) => *slot,
+        }
+    }
+
+    /// Is this a fast finalize certificate?
+    pub fn is_fast_finalization(&self) -> bool {
+        matches!(self, Self::FinalizeFast(_, _))
+    }
+
+    /// Is this a finalize / fast finalize certificate?
+    pub fn is_finalization(&self) -> bool {
+        matches!(self, Self::Finalize(_) | Self::FinalizeFast(_, _))
+    }
+
+    /// Is this a notarize fallback certificate?
+    pub fn is_notarize_fallback(&self) -> bool {
+        matches!(self, Self::NotarizeFallback(_, _))
+    }
+
+    /// Is this a skip certificate?
+    pub fn is_skip(&self) -> bool {
+        matches!(self, Self::Skip(_))
+    }
+
+    /// Gets the block associated with this certificate, if present
+    pub fn to_block(self) -> Option<Block> {
+        match self {
+            Certificate::Finalize(_) | Certificate::Skip(_) => None,
+            Certificate::Notarize(slot, block_id)
+            | Certificate::NotarizeFallback(slot, block_id)
+            | Certificate::FinalizeFast(slot, block_id) => Some((slot, block_id)),
+        }
+    }
+
+    /// "Critical" certs are the certificates necessary to make progress
+    /// We do not consider the next slot for voting until we've seen either
+    /// a Skip certificate or a NotarizeFallback certificate for ParentReady
+    ///
+    /// Note: Notarization certificates necessarily generate a
+    /// NotarizeFallback certificate as well
+    pub fn is_critical(&self) -> bool {
```

Comment:
nit: I don't think you need this any more after we remove cert repair, but it's okay to do it in another PR.